src/createHeadlessForm.js

```diff
@@ -323,9 +324,10 @@ export function createHeadlessForm(jsonSchema, customConfig = {}) {
   };
 
   try {
-    const fields = getFieldsFromJSONSchema(jsonSchema, config);
+    const validations = createValidationChecker(jsonSchema);
+    const fields = getFieldsFromJSONSchema(jsonSchema, config, validations);
 
-    const handleValidation = handleValuesChange(fields, jsonSchema, config);
+    const handleValidation = handleValuesChange(fields, jsonSchema, config, validations);
 
     updateFieldsProperties(fields, getPrefillValues(fields, config.initialValues), jsonSchema);
```

Comment:
Shouldn't we also pass the `validations` param to the `updateFieldsProperties`?

Reply:
As per above comment, I'm explicitly avoiding this til later PRs where its all tackled! This was purely to keep the code line count down as much as possible and spotlight those changes in the other PRs.